Merge Development for new release

aschiffler opened the pull request:
This merge will have no major improvements in the release but some small fixes. Most in the Web UI.
Target release 0.5.16 on Sunday 9/4/22

DanielR92 committed:
Delete checkserial() in the input field

Webbrowser error when the site 'setup' is called. 
JS function not exist.

https://github.com/grindylow/ahoy/issues/184 (first row)

DanielR92 committed:
Update Readme start

Start to review the whole documentation and rewrite it. 
First draft. :)

DanielR92 committed:
Update first draft

added the todo list from the post https://github.com/grindylow/ahoy/issues/142

DanielR92 committed:
updated the ESP8266 readme

aschiffler committed:
Merge pull request #187 from DanielR92/patch-5

Delete checkserial() in the input field

aschiffler committed:
Merge pull request #188 from DanielR92/patch-6

Update Readme start

aschiffler committed:
Merge pull request #189 from DanielR92/patch-7

Update first draft

tastendruecker123 committed:
MQTT reconnect delay, fixed ESP32 build

lumapu committed:
Merge pull request #194 from tastendruecker123/dev2

MQTT reconnect delay, fixed ESP32 build

aschiffler committed:
added last alarm msg request

aschiffler committed:
added esp32 to release work flow

aschiffler committed:
improvment mqtt after add values

DanielR92 committed:
Added a Hint for update from a low version to new

DanielR92 committed:
Update Scheduler / Task manager + MQTT

aschiffler committed:
Merge pull request #199 from DanielR92/patch-8

Added a Hint for update from a low version to new

aschiffler committed:
Merge pull request #200 from DanielR92/patch-10

Update Scheduler / Task manager + MQTT

lumapu committed:
code review: added check if mqtt is enabled before sending out values

lumapu committed:
improved generation of power limit control html select element

lumapu committed:
* fix empty inverters save on a brandnew ESP
* replaced several {0} by memset
* erase now erases the flash with 0xff (identical to a brandnew ESP)

lumapu committed:
* added option for no powerlimit (setup)
* moved function showLiveData to web.cpp
* improved debug messages
* cleaned some code

aschiffler committed:
add github actions for dev builds

Build automatically the firmware under tools/esp8266 for any branchs whichs matchs the widlcard development* and plces the binaries in the action artifacts

aschiffler committed:
Update compile_development.yml

aschiffler committed:
fix development workflow actions

aschiffler committed:
Merge branch 'development02' of https://github.com/grindylow/ahoy into development02

cubinet-code committed:
OpenScad parametric Case for Dollatek NRF24L01

cubinet-code committed:
- Improve config system allowing a user customizable config_override.h file to ensure git pulls don't require merges
- Move DTU_RADIO_ID to config.h to allow for easy customization for multiple DTUs with non conflicting radio IDs

aschiffler committed:
Merge branch 'development02' into config_override

aschiffler committed:
Merge pull request #211 from cubinet-code/config_override

Improve config system and make DTU_RADIO_ID customizable in config.h

cubinet-code committed:
Clarify photo file names

lumapu committed:
Merge branch 'main' into development02

lumapu committed:
* fix #195: used trim() to remove leading and trailing spaces, added maxlength attr to input field
* started to work on #209: added another timer for NTP update after configurable interval (default 12h)

lumapu committed:
fix #209

lumapu committed:
fix action (directory was missing - needs to be created automatically now)

lumapu committed:
badge test

thfcm committed:
Update README.md

Added Discord Link.

aschiffler committed:
Merge branch 'development02' into patch-2

aschiffler committed:
Merge pull request #218 from thfcm/patch-2

Updates README.md (thx @thfcm )

aschiffler committed:
change dir for cases

aschiffler committed:
Merge branch 'dollatek_case' into development02

aschiffler committed:
Update User_Manual.md

lumapu commented on tools/esp8266/README.md:
```diff
@@ -28,6 +28,23 @@ Additional there are 3 pins, which can be set individual:
+- HM400
+- HM600
+- HM700
+- HM800
```
add HM1000?

lumapu commented on tools/esp8266/README.md:
```diff
@@ -36,63 +53,95 @@ This code can be compiled using Visual Studio Code and **PlatformIO** Addon. The
 - Board: Generic ESP8266 Module
```
remove

lumapu commented on tools/esp8266/README.md:
```diff
@@ -36,63 +53,95 @@ This code can be compiled using Visual Studio Code and **PlatformIO** Addon. The
+  - `RF24` 1.4.5
+  - `PubSubClient` 2.8
+  - `ArduinoJson` 6.19.4
+  
```
remove, these are automatically installed

lumapu commented on tools/esp8266/app.cpp:
```diff
@@ -53,18 +53,22 @@ void app::loop(void) {
     mWebInst->loop();
 
     if(checkTicker(&mUptimeTicker, mUptimeInterval)) {
```
this must be removed because the next line, we don't have to compare millis twice

lumapu commented on tools/esp8266/html/setup.html:
```diff
@@ -102,7 +102,7 @@ <h1>Setup</h1>
                     <legend class="des">MQTT</legend>
                         <label for="mqttAddr">Broker / Server IP</label>
-                        <input type="text" class="text" name="mqttAddr" value="{MQTT_ADDR}"/>
+                        <input type="text" class="text" name="mqttAddr" value="{MQTT_ADDR}" maxlength="32" />
```
the maxlengh should come from define and not hard coded

aschiffler replied:
Thx for review

aschiffler replied:
Won't fix that for now, bc maybe it will be solved by reworked html/webserver things.

lumapu committed:
Merge branch 'main' into development02

lumapu committed:
corrected most of the review comments, except "maxlength"

thfcm committed:
Update README.md

Überarbeitete Readme.md
- Umstrukturierung der Themen
- Aktualisierte Inhaltsliste
- Ergänzungen

thfcm committed:
Update README.md

Implemented Change Request

lumapu committed:
Merge pull request #232 from thfcm/patch-4

Update README.md

aschiffler committed:
Update Changes for release 0.5.16
